fferr/generic.go

```diff
@@ -44,6 +44,32 @@ func (e *GenericError) AddDetail(key, value string) {
 	e.detailKeys = append(e.detailKeys, key)
 }
 
+func (e *GenericError) AddDetails(keysAndValues ...interface{}) {
+	// Modeled after the logging.infow method
+
+	if len(keysAndValues)%2 != 0 {
+		// We're just going to print an error if the number of arguments is odd and omit the last one.
+		fmt.Println("AddDetails called with odd number of arguments, expected key-value pairs")
+	}
+
+	n := len(keysAndValues)
+	if n%2 != 0 {
+		n = n - 1 // Skip the last one if odd
+	}
+
+	for i := 0; i < n; i += 2 {
+		key, ok := keysAndValues[i].(string)
+		if !ok {
+			// Handle or log the error if the key is not a string.
+			fmt.Printf("Key at index %d is not a string\n", i)
+			continue
+		}
+		// Convert the value to a string, this assumes that the value has a meaningful string representation
```

Comment:
I don't think we should be making this assumption. I think we should force the consumer to use a string value so it's clear to them that without a meaningful string representation, they can't add details.